core/trino-main/src/main/java/io/trino/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -135,6 +144,82 @@ public static PlanNodeStatsEstimate capStats(PlanNodeStatsEstimate stats, PlanNo
         return result.build();
     }
 
+    public static Map<Symbol, SymbolStatsEstimate> intersectCorrelatedStats(List<PlanNodeStatsEstimate> estimates)
+    {
+        checkArgument(!estimates.isEmpty(), "estimates is empty");
+        if (estimates.size() == 1) {
+            return estimates.get(0).getSymbolStatistics();
+        }
+        PlanNodeStatsEstimate.Builder result = PlanNodeStatsEstimate.builder();
+        // Update statistic range for symbols
+        estimates.stream().flatMap(estimate -> estimate.getSymbolsWithKnownStatistics().stream())
+                .distinct()
+                .forEach(symbol -> {
+                    List<SymbolStatsEstimate> symbolStatsEstimates = estimates.stream()
+                            .map(estimate -> estimate.getSymbolStatistics(symbol))
+                            .collect(toImmutableList());
+
+                    StatisticRange intersect = symbolStatsEstimates.stream()
+                            .map(StatisticRange::from)
+                            .reduce(StatisticRange::intersect)
+                            .orElseThrow();
+
+                    // intersectCorrelatedStats should try produce stats as if filters are applied in sequence.
+                    // Using min works for filters like (a > 10 AND b < 10), but won't work for
+                    // (a > 10 AND b IS NULL). However, former case is more common.
```

Comment:
nit: `but won't work for (a > 10 AND b IS NULL)`
The test actually shows that it works for that case too... right?

Reply:
it happens to be corrected by stats normaliser, the output of this method follows the comment